### PR TITLE
[CDAP-20000] Logging ProgramRunId in case of ProgramDispatch failure

### PR DESCRIPTION
**Pull request by adityagupta94**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/ProgramRunDispatcherTask.java [hunk 1]
```diff
@@ -68,13 +68,18 @@ public void run(RunnableTaskContext context) throws Exception {
...
+    ProgramRunId programRunId =
+      dispatcherContext.getProgramDescriptor().getProgramId().run(dispatcherContext.getRunId());
 
     ProgramRunDispatcher dispatcher = injector.getInstance(InMemoryProgramRunDispatcher.class);
-    ProgramController programController = dispatcher.dispatchProgram(dispatcherContext);
+    ProgramController programController;
+    try {
+      programController = dispatcher.dispatchProgram(dispatcherContext);
+    } catch (Exception e) {
+      throw new Exception("Failure while dispatching program run " + programRunId, e);
```
Comment (masoud-io) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/ProgramRunDispatcherTask.java]: Use String.format

Comment (masoud-io) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/ProgramRunDispatcherTask.java]: nit: change to `Failed to dispatch program run`

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/ProgramRunDispatcherTask.java [hunk 2]
```diff
@@ -68,13 +68,18 @@ public void run(RunnableTaskContext context) throws Exception {
     ProgramRunDispatcherContext dispatcherContext = new ProgramRunDispatcherContext(
       GSON.fromJson(context.getParam(), ProgramRunDispatcherContext.class));
     context.setCleanupTask(dispatcherContext::executeCleanupTasks);
+    ProgramRunId programRunId =
```
Comment (masoud-io) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/ProgramRunDispatcherTask.java:71]: We don't need to initialize programRunId here. Can be moved to the catch block.

**Review by albertshau — APPROVED** [review]

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/ProgramRunDispatcherTask.java (hunk 1 above)
Comment (adityagupta94) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/ProgramRunDispatcherTask.java]: Done

Comment (adityagupta94) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/ProgramRunDispatcherTask.java]: Done

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/ProgramRunDispatcherTask.java (hunk 2 above)
Comment (adityagupta94) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/ProgramRunDispatcherTask.java:71]: programRunId was required outside of catch block also. Hence declared it above the try-catch.

**Pull request merged**